deepspeed/runtime/engine.py

```diff
@@ -2467,6 +2480,12 @@ def load_checkpoint(self,
         ``load_checkpoint()`` wants a pristine model. If insisting to do so, please reinitialize engine
         before ``load_checkpoint()``.
         """
+        checkpoint_engine_tmp = self.checkpoint_engine
```

Comment:
I don't think this temporary swap of checkpoint_engine is needed here since NebulaCheckpointEngine will be initialized by the client through deepspeed configuration. Is that correct?

Reply:
I want to confirm that after moving the args to config as discussed that this change will go away.

Reply:
Yep. Thanks for confirming. Removed now.